app/commands/process_raw_event.rb

```diff
@@ -0,0 +1,104 @@
+class ProcessRawEvent
+  def self.call(*args)
+    new(*args).call
+  end
+
+  def initialize(raw_event_id)
+    @raw_event_id = raw_event_id
+  end
+
+  def call
+    raw_event = RawEvent.find_by(id: @raw_event_id)
+    return unless raw_event
+
+    process_raw_event(raw_event)
+  end
+
+  private
+
+  def process_raw_event(raw_event)
+    event_source = find_or_create_event_source(raw_event)
+    scenario = find_or_create_scenario(
+      event_source: event_source,
+      hash: generate_scenario_hash(raw_event.details)
+    )
+    raw_event.update(scenario: scenario)
+
+    store_request(raw_event)
+    store_events(raw_event)
+  end
+
+  def generate_scenario_hash(details)
+    scenario_key = details.map { |d| d[:type] }.join
+    Scenario.hash_from_string(scenario_key)
+  end
+
+  def find_or_create_event_source(raw_event)
+    application = raw_event.application
+
+    application.event_sources.find_or_create_by(
+      source: raw_event.meta[:controller],
+      endpoint: raw_event.meta[:action]
+    )
+  end
+
+  def find_or_create_scenario(event_source:, hash:)
+    event_source.scenarios.find_or_create_by(events_hash: hash)
+  end
+
+  def store_request(raw_event)
+    request = Request.new(request_hash(raw_event)).to_json
+
+    Commands::SendToKafka.call request
+  end
+
+  def store_events(raw_event)
+    events = raw_event.details.each_with_index.map do |row, index|
+      Event.new(
+        event_hash(scenario: raw_event.scenario, details: row, meta: raw_event.meta, index: index)
+      ).to_json
+    end
+
+    Commands::SendToKafka.call events
+  end
+
+  def request_hash(raw_event)
+    scenario = raw_event.scenario
+    event_source = scenario.event_source
+    meta = raw_event.meta
+
+    {
+      timestamp: Time.at(meta[:start]).to_s(:iso8601),
```

Comment:
Do not use `Time.at` without zone. Use one of `Time.zone.at`, `Time.at.current`, `Time.at.in_time_zone`, `Time.at.utc`, `Time.at.getlocal`, `Time.at.iso8601`, `Time.at.jisx0301`, `Time.at.rfc3339`, `Time.at.to_i`, `Time.at.to_f` instead.